### PR TITLE
Create swc plugin that removes prop types in production builds

### PR DESCRIPTION
**Pull request by axbarsan**
This PR adds a plugin for the `swc` compiler, which strips react component `propTypes` and removes the `prop-types` import when building the production app. We don't need them because they don't run in production anyway, so they just take up extra code.

The savings are not noticeable (~40 KB), but it was an occasion for me to have some fun with `swc`.

Not all possible scenarios are handled (you can see what's handled in the `PropTypesStripper` test), but all the patterns that we use right now work.
In order to make my job easier, I've changed the code that was not following our current pattern, to have everything working smoothly. 👍🏻

### CODE REVIEW

File: scripts/swc/__tests__/PropTypesStripper.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { parse, parseSync, transformSync } from '@swc/core';
+import PropTypesStripper from '../PropTypesStripper';
+
+describe('PropTypesStripper', () => {
```
Comment (axbarsan) [inline scripts/swc/__tests__/PropTypesStripper.ts:4]: Plugin test cases

File: scripts/swc/PropTypesStripper.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  ClassDeclaration,
+  Declaration,
+  Expression,
+  ExpressionStatement,
+  ImportDeclaration,
+  Module,
+  Program,
+} from '@swc/core';
+
+class PropTypesStripper {
```
Comment (axbarsan) [inline scripts/swc/PropTypesStripper.ts:12]: Plugin

Comment (axbarsan) [reply scripts/swc/PropTypesStripper.ts:12]: Visits certain AST nodes and replaces them with other AST nodes

**Review by marians — APPROVED** [review]
Every byte counts :)

**Pull request merged**